src/topology/uniform_space/abstract_completion.lean

```diff
@@ -222,19 +223,28 @@ begin
   refl
 end
 
-/-- The bijection between two completions of the same uniform space. -/
-def compare_equiv : pkg.space ≃ pkg'.space :=
+/-- The uniform bijection between two completions of the same uniform space. -/
+def compare_equiv : pkg.space ≃ᵤ pkg'.space :=
 { to_fun := pkg.compare pkg',
   inv_fun := pkg'.compare pkg,
   left_inv := congr_fun (pkg'.inverse_compare pkg),
-  right_inv := congr_fun (pkg.inverse_compare pkg') }
+  right_inv := congr_fun (pkg.inverse_compare pkg'),
+  uniform_continuous_to_fun := uniform_continuous_compare _ _,
+  uniform_continuous_inv_fun := uniform_continuous_compare _ _, }
 
 lemma uniform_continuous_compare_equiv : uniform_continuous (pkg.compare_equiv pkg') :=
 pkg.uniform_continuous_compare pkg'
 
 lemma uniform_continuous_compare_equiv_symm : uniform_continuous (pkg.compare_equiv pkg').symm :=
 pkg'.uniform_continuous_compare pkg
 
+/-- The uniform bijection between a complete space and its abstract completion. -/
+def complete_equiv_self {α : Type*} [uniform_space α]
+[separated_space α] [complete_space α] (pkg : abstract_completion α) :
+pkg.space ≃ᵤ α :=
+compare_equiv pkg (mk α id(by apply_instance) (by apply_instance) (by apply_instance)
+  uniform_inducing_id dense_range_id)
+
```

Comment:
I think you should extract the key fact here, which is the fact that `α` is an abstract completion over `α`. Then I'd say we only need the concrete version `uniform_completion.complete_equiv_self`, because the whole point of the `abstract_completion` setup is that we *don't* have to duplicate the API, we just have to provide an abstract completion package. I hope that makes sense :sweat_smile:

Reply:
Anatole, do you mean I should prove that ```α``` is an instance of ```abstract_completion α``` when it is complete and use that to prove the ```uniform_completion.complete_equiv_self``` (and thus remove ```abstract_completion.complete_equiv_self```)? That makes sense to me.

Reply:
Yes exactly!

Reply:
Ok. I have committed the new version.